src/basic/terminal-util.c

```diff
@@ -1272,12 +1272,16 @@ ColorMode get_color_mode(void) {
 
         if (cached_color_mode < 0) {
                 m = parse_systemd_colors();
+                const char *colorterm = getenv("COLORTERM");
                 if (m >= 0)
                         cached_color_mode = m;
                 else if (getenv("NO_COLOR"))
                         /* We only check for the presence of the variable; value is ignored. */
                         cached_color_mode = COLOR_OFF;
 
+                else if (colorterm != NULL && (streq(colorterm, "truecolor") || streq(colorterm, "24bit")))
```

Comment:
you can use `streq_ptr` and drop the manual null check

Reply:
… or even use `STRPTR_IN_SET` !

Reply:
We don't want to call `getenv` unless we're going to use the return value. It needs to search the array of variables, so it is "expensive".

So this should be like this:
```C
... else if (STRPTR_IN_SET(getenv("COLORTERM"),
               "truecolor",
               "24bit")) 

        ...
```

Reply:
We don't want to call `getenv` unless we're going to use the return value. It needs to search the array of variables, so it is "expensive".

So this should be like this:
```C
     ...
     else if (STRPTR_IN_SET(getenv("COLORTERM"),
               "truecolor",
               "24bit")) 

        ...
```

Reply:
Thanks for sharing the util. Updated the commit. I will though understand how STRPTR_IN_SET or strreq_ptr better suited here.